shared-bindings/keypad/KeyMatrix.c

```diff
@@ -90,6 +90,10 @@ STATIC mp_obj_t keypad_keymatrix_make_new(const mp_obj_type_t *type, size_t n_ar
     mcu_pin_obj_t *row_pins_array[num_row_pins];
     mcu_pin_obj_t *column_pins_array[num_column_pins];
 
+    validate_no_duplicate_pins(row_pins, MP_QSTR_row_pins);
+    validate_no_duplicate_pins(column_pins, MP_QSTR_column_pins);
+    validate_no_duplicate_pins_2(row_pins, column_pins, MP_QSTR_row_pins, MP_QSTR_column_pins);
```

Comment:
Instead of calling three routines here, suppose `validate_no_duplicate_pins_2()` called `validate_no_duplicate_pins()` on each of the lists it was passed. I can't think of a case where it's OK to have duplicates between the lists. So then a single call does all the validation. It doesn't make a difference in this PR but if `validate_no_duplicate_pins_2()` were called elsewhere it would save some calls.

Reply:
Makes sense to me. Changed it.